dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -302,13 +307,50 @@ def __init__(
         self.active_hints = deepcopy(HINT_TO_CLICKHOUSE_ATTR)
         self.type_mapper = ClickHouseTypeMapper(self.capabilities)
 
+    def has_dataset(self) -> bool:
```

Comment:
Not sure if this is the correct way to interact with state. I think we need @rudolfix to review this block.

Reply:
Yeah I've a bit stumped here. Thanks @rudolfix

Reply:
This test seems to be passing intermittently.